ynput/ayon-launcher
# Develop mode

## Changelog Description
Implemented support of develop mode in launcher. The launcher can skip distribution of addons by installing to addons dir, instead can use paths from server. And added disk remapping functionality.

## Additional info
Added new flag argument `--use-dev` which will find bundle that is marked as dev bundle for the user who is logged in. It is also possible to start dev mode by passing bundle name which is set as dev bundle, in that case it does not check if user is assigned to it and blindly use it (handy for farm etc.).

Because settings variant is not as easy to define as used to be, it happens in the middle of distribution > that is because distribution is already preparing all the data needed for distribution itself.

Staging is completelly ignored if dev bundle should be used.

To be able to fully test the functionality it is necessary to have `openpype` addon with changes in this PR https://github.com/ynput/OpenPype/pull/5783 .

## Testing notes:
1. Make sure you update venv (`./tools/manage.ps1 create-env` | `./tools/make.sh create-env`)
2. Make sure you have latest code of `openpype` addon on server from this PR https://github.com/ynput/OpenPype/pull/5783
3. Make a build (or run from code)
4. Enabled Develop mode on server (user must have enabled "is developer")
5. Create develop bundle
6. Assign your user to the bundle
7. In the bundles tab, select the bundle, find `openpype` addon and enable dev path, and fill the path with path to openpype repository.
    - this should be possible to do with any other addon too, but with this you can test https://github.com/ynput/OpenPype/pull/5783 without uploading it to server
8. Run launcher with `./ayon_console.exe --use-dev` or `./ayon_console.exe --bundle <your dev bundle name>`
9. The addons from custom directories should be used
